### PR TITLE
[Klaviyo] Update profile identifiers for track and order completed events.

### PR DESCRIPTION
**Pull request by harsh-joshi99**
This PR adds `external_id` and `anonymous_id` as profile identifiers, removes `other_properties` nested object from profile.

JIRA -> [STRATCONN-3857](https://segment.atlassian.net/browse/STRATCONN-3857)

## Testing
Mappings:
Screenshot 2024-06-10 at 5 44 34 PM

**Anonymous Id**: 

Track Event:
Screenshot 2024-06-10 at 5 40 58 PM
Screenshot 2024-06-10 at 5 41 47 PM

Order Completed Event:
Screenshot 2024-06-10 at 6 22 31 PM
Screenshot 2024-06-10 at 6 22 52 PM

**External Id**:
Track Event:
Screenshot 2024-06-10 at 5 43 00 PM
Screenshot 2024-06-10 at 5 43 18 PM

Order Completed Event:
Screenshot 2024-06-10 at 6 38 01 PM

Email:
Track Event:
Screenshot 2024-06-10 at 5 46 03 PM
Screenshot 2024-06-10 at 5 46 16 PM

Order Completed Event:
Screenshot 2024-06-10 at 7 14 43 PM
Screenshot 2024-06-10 at 7 14 37 PM

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [x] [Segmenters] Tested in the staging environment

[STRATCONN-3857]: https://segment.atlassian.net/browse/STRATCONN-3857?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/destination-actions/src/destinations/klaviyo/orderCompleted/index.ts [hunk 1]
```diff
@@ -88,11 +86,21 @@ const action: ActionDefinition<Settings, Payload> = {
           label: 'Phone Number',
           type: 'string'
```
Comment (AnkitSegment) [inline packages/destination-actions/src/destinations/klaviyo/orderCompleted/index.ts:87]: Can we add required: true here ?

File: packages/destination-actions/src/destinations/klaviyo/orderCompleted/index.ts [hunk 2]
```diff
@@ -88,11 +86,21 @@ const action: ActionDefinition<Settings, Payload> = {
...
         },
-        other_properties: {
-          label: 'Other Properties',
-          type: 'object'
+        external_id: {
+          label: 'External Id',
+          description:
+            'A unique identifier used by customers to associate Klaviyo profiles with profiles in an external system.',
+          type: 'string',
+          default: { '@path': '$.userId' }
```
Comment (AnkitSegment) [inline packages/destination-actions/src/destinations/klaviyo/orderCompleted/index.ts:94]: Can we add required: true here ?

File: packages/destination-actions/src/destinations/klaviyo/orderCompleted/index.ts [hunk 3]
```diff
@@ -88,11 +86,21 @@ const action: ActionDefinition<Settings, Payload> = {
...
+        },
+        anonymous_id: {
+          label: 'Anonymous Id',
+          description: 'Anonymous user identifier for the user.',
+          type: 'string',
+          default: { '@path': '$.anonymousId' }
```
Comment (AnkitSegment) [inline packages/destination-actions/src/destinations/klaviyo/orderCompleted/index.ts:100]: Can we add required: true here ?

**Review by AnkitSegment — COMMENTED** [review]
Looks good! Can we add required parameter?

**Review by varadarajan-tw — APPROVED** [review]

File: packages/destination-actions/src/destinations/klaviyo/orderCompleted/index.ts (hunk 3 above)
Comment (harsh-joshi99) [reply packages/destination-actions/src/destinations/klaviyo/orderCompleted/index.ts:100]: Only one of these identifier fields are required and that validation check is added.

File: packages/destination-actions/src/destinations/klaviyo/orderCompleted/index.ts (hunk 2 above)
Comment (harsh-joshi99) [reply packages/destination-actions/src/destinations/klaviyo/orderCompleted/index.ts:94]: Only one of these identifier fields are required and that validation check is added.

File: packages/destination-actions/src/destinations/klaviyo/orderCompleted/index.ts (hunk 1 above)
Comment (harsh-joshi99) [reply packages/destination-actions/src/destinations/klaviyo/orderCompleted/index.ts:87]: Only one of these identifier fields are required and that validation check is added.

**Pull request merged**